File: backend/src/main/java/com/poshanforlife/api/dto/UpdateUserRequest.java
package com.poshanforlife.api.dto;

import com.poshanforlife.api.entity.Role;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;

/**
 * PATCH body — all fields optional; absent (null) fields are left unchanged.
 * name/phone/fcmToken: self or admin. role/isActive/dateOfBirth: admin only —
 * a non-admin sending them gets 422 VALIDATION_ERROR (rejected, not ignored).
 * fcmToken is sent by the Android app on login and on FirebaseMessagingService
 * .onNewToken(); an empty string clears it (e.g. on logout).
 */
public record UpdateUserRequest(
        @Size(min = 2, max = 255) String name,
        @Size(max = 32) String phone,
        Role role,
        Boolean isActive,
        LocalDate dateOfBirth,
        @Size(max = 255) String fcmToken) {
}

File: android/app/src/main/java/com/poshanforlife/android/core/network/UpdateUserRequest.kt
package com.poshanforlife.android.core.network

import kotlinx.serialization.Serializable

/**
 * All fields optional — absent (null) ones are left unchanged server-side
 * (explicitNulls = false means a null field is omitted from the request
 * body entirely, not sent as literal JSON null). Only fcmToken is used by
 * this app today (name/phone edits aren't built yet); other backend-side
 * fields (role/isActive/dateOfBirth) are admin-only and irrelevant here.
 */
@Serializable
data class UpdateUserRequest(
    val name: String? = null,
    val phone: String? = null,
    val fcmToken: String? = null,
)
